packages/common/src/index.ts

```diff
@@ -549,7 +552,7 @@ export default class Common extends EventEmitter {
    * @returns True if HF is active on block number
    */
   hardforkIsActiveOnBlock(
-    hardfork: string | null,
+    hardfork: string | Hardforks | null,
```

Comment:
Hmm, I have to say for typings this plural version really not looks so nice but is rather unintuitive. This seems a lot better to grasp if it would be:

```typescript
hardfork: string | Hardfork | null
```

Maybe we want to switch back to singular, also for the `CustomChains` enum I have introduced (Emmett, could you do here if we decide to do?).

For the `Chain` interface I guess we can just make a renamed import with `as`? The `Chain` interface is also likely not that-much used by end users, so this shouldn't be such a problem, and if someone uses it it is also still possible to do a renamed import.

@ryanio @emersonmacro what do you guys think? 🤔

Reply:
(I am slowly getting to the point to think that I am close to always circle in back after 3-4 iterations to the things Ryan proposed initially. Maybe I should generally directly move over and leave the in-between iterations) to safe some time here. 😂)

Reply:
I have to say, I went through a similar progression. At first, I thought the plural would work well, but the more I saw it in the code, the more I thought the singular seemed more natural. Updated throughout, and updated `CustomChains` to `CustomChain` as well.

For the interface, I went with `import { Chain as IChain } from './types'`, and then used `IChain` throughout `index.ts`. Let me know if you think we should use something else for the `as`.